consensus/src/round_manager.rs

```diff
@@ -696,6 +708,23 @@ impl RoundManager {
         sync_or_not
     }
 
+    fn compute_timeout_reason(&self, round: Round) -> RoundTimeoutReason {
+        if self.round_state().vote_sent().is_some() {
+            return RoundTimeoutReason::NoQC;
+        }
+
+        match self.block_store.get_block_for_round(round) {
+            None => RoundTimeoutReason::ProposalNotReceived,
+            Some(block) => {
+                if let Err(missing_authors) = self.block_store.check_payload(block.block()) {
```

Comment:
hmm we still insert the block if the payload is unavailable?

Reply:
Yes, we just don't vote on it.

Reply:
if the payload is not available we start to fetch the data? do we set up timeout (to be the same as round timeout) for the fetch?

Reply:
Yes, we do. There was a timeout of 1 second, but I just changed it to match the round timeout.